### PR TITLE
Add a cache clear method to the UtilsService

### LINKED ISSUE #19958: Add clear-cache method to UtilsService
Most endpoints under the utils endpoints use the utils service to run the operations, except for clear-cache. This endpoint has the cache clearing "hardcoded" in, rather than using the service.

We should refactor that to use the UtilsService as well, which in turn also allows extensions to clear the cache.

Ref https://github.com/directus/directus/discussions/19945

### PR DESCRIPTION
**Pull request by nodeworks**
## Scope

What's changed:

- Added a clear cache method to the UtilsService

## Potential Risks / Drawbacks

- None

---

Resolves #19958

### CODE REVIEW

**Review by paescuj — APPROVED** [review]

File: api/src/services/utils.ts [hunk 1]
```diff
@@ -147,4 +147,12 @@ export class UtilsService {
 			}
 		);
 	}
+
+	async clearCache(): Promise<void> {
+		if (this.accountability?.admin !== true) {
+			throw new ForbiddenError();
+		}
```
Comment (br41nslug) [inline api/src/services/utils.ts:154]: I think perhaps we should keep this check in the controller only so it is not required to fake an accountability object (when running from a crontab event for instance)

**Review by br41nslug — COMMENTED** [review]
Looking good, just that one suggestion

File: api/src/services/utils.ts (hunk 1 above)
Comment (paescuj) [reply api/src/services/utils.ts:154]: Wasn't quite sure at first either, but I think the check makes more sense as part of the service, as an extension might get `accountability` from a request as well and pass it along to the service. Feels safer and also it's consistent with other service methods where the permission check takes place at that level too.

Comment (rijkvanzanten) [reply api/src/services/utils.ts:154]: I think it's fine to have that check here. Clearing the cache is admin only; in things like flows and operations there is additional options to "Run as Admin", so we should have things covered

Comment (rijkvanzanten) [reply api/src/services/utils.ts:154]: It's the same as with every other system service 👍🏻

Comment (paescuj) [reply api/src/services/utils.ts:154]: > [...] in things like flows and operations there is additional options to "Run as Admin", so we should have things covered

Good point 👍🏻

**Pull request merged**